app/scripts/modules/core/src/managed/displayNames.ts

```diff
@@ -0,0 +1,7 @@
+import { IManagedResourceSummary, IManagedArtifactVersion } from '../domain';
+
+export const getResourceName = ({ moniker: { app, stack, detail } }: IManagedResourceSummary) =>
+  [app, stack, detail].filter(Boolean).join('-');
+
+export const getArtifactVersionDisplayName = ({ displayName, build, git }: IManagedArtifactVersion) =>
+  build?.id ? `#${build?.id}` : git?.commit || displayName || 'unknown';
```

Comment:
I don't know if I really care for this file's taxonomy/naming much, but we needed a good spot to share these functions and I don't think I'm quite ready to go touch `core/naming` yet 😬